src/DependencyInjection/Compiler/EventHandlerCompilerPass.php

```diff
@@ -29,6 +33,14 @@ public function process(ContainerBuilder $container)
         foreach ($taggedServices as $id => $tags) {
             foreach ($tags as $attributes) {
                 if (isset($attributes['service']) && in_array($attributes['event'], GuzzleEvents::EVENTS, true)) {
+                    @trigger_error(
```

Comment:
You already added the deprecation message about GuzzleEventListenerInterface directly in the class of interface. Do we need this special message in compiler pass?

Reply:
I think so, because people may be using it for different reasons, and since the interface will be removed in v8 all usages of that interface need a deprecation notice.